app/models/home_feed.rb

```diff
@@ -9,4 +9,40 @@ def initialize(account)
   def regenerating?
     redis.exists?("account:#{@account.id}:regeneration")
   end
+
+  def get(limit, max_id = nil, since_id = nil, min_id = nil)
+    limit    = limit.to_i
+    max_id   = max_id.to_i if max_id.present?
+    since_id = since_id.to_i if since_id.present?
+    min_id   = min_id.to_i if min_id.present?
+
+    redis_min_id = from_redis(1, nil, nil, 0).first&.id
+
+    statuses = from_redis(limit, max_id, since_id, min_id)
```

Comment:
A minor optimization could be to do `statuses = from_redis(limit, max_id, since_id, min_id)` first and early return if `statuses.size >= limit`. Also only fetching `redis_min_id` if `min_id` or `since_id` have been passed.